src/components/EditorWrapper.js

```diff
@@ -1,123 +1,129 @@
-import React, { useEffect, useRef, useState } from 'react';
+import React, { useEffect, useRef, useState } from "react";
 import Editor, { DiffEditor } from "@monaco-editor/react";
 
 const cleanEditorContent = {
-    title: "index",
-    value: "",
-    inputType: "json"
-}
+  title: "index",
+  value: "",
+  inputType: "json",
+};
 
 const defaultOptions = { readOnly: true, domReadOnly: true };
 
 function EditorWrapper({ results }) {
-    const [editorContent, setEditorContent] = useState(cleanEditorContent);
-    const [editorOptions, setEditorOptions] = useState(defaultOptions);
+  const [editorContent, setEditorContent] = useState(cleanEditorContent);
+  const [editorOptions, setEditorOptions] = useState(defaultOptions);
+  const [cacheKey, setCacheKey] = useState(1);
 
-    const editorRef = useRef(null);
+  const editorRef = useRef(null);
 
-    useEffect(() => {
-        console.log("updated editor");
-        editorRef.current?.focus();
-    }, [editorContent?.id]);
+  useEffect(() => {
+    editorRef.current?.focus();
+  }, [editorContent?.id]);
 
-    const firstResult = results[0]?.id
+  const firstResult = results[0]?.id;
 
-    useEffect(() => {
-        if (!editorContent.id && firstResult) setEditorContent(results[results.length - 1])
-        if (!results.length && editorContent.id) setEditorContent(cleanEditorContent);
-    }, [firstResult, editorContent, results])
-    console.log({editorContent})
+  useEffect(() => {
+    if (!editorContent.id && firstResult)
+      setEditorContent(results[results.length - 1]);
+    if (!results.length && editorContent.id) {
+      setEditorContent(cleanEditorContent);
+      setCacheKey((prev) => prev + 1);
```

Comment:
@DanielC-N I think this is the magic cache-fixing line.

Reply:
![](https://i.giphy.com/media/1jnyRP4DorCh2/giphy.webp)